libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -204,9 +204,17 @@ bool AP_Arming::airspeed_checks(bool report)
             return true;
         }
         for (uint8_t i=0; i<AIRSPEED_MAX_SENSORS; i++) {
-            if (airspeed->enabled(i) && airspeed->use(i) && !airspeed->healthy(i)) {
-                check_failed(ARMING_CHECK_AIRSPEED, report, "Airspeed %d not healthy", i + 1);
-                return false;
+            if (airspeed->enabled(i)) {
```

Comment:
This should check that it is set for using this sensor.

Reply:
I do not agree with this point. I've explained the reason in the head. 
```
Always enable checks if airspeed is enabled, even if not `used`. Other sensors checks works like this. 
Otherwise user can knowingly disable airspeed. Also, airspeed is used on crash detection and dead-reckoning on GPS loss even if not 'used'.
```
If a user doesn't need to check for airspeed. He can knowingly can either disable airspeed or disable the check.

Other sensors are checked even if not conditionally used.

Reply:
Also, I remember that if airspeed is enable `even not used by parameter` it's used for wind estimation which helps a lot in maneuvering. Also it's used if just enabled in landing approach.
Correct me if I'm wrong here.

Reply:
> Other sensors checks works like this. 

Can you give an example of such sensor?

> If a user doesn't need to check for airspeed. He can knowingly can either disable airspeed or disable the check.

No, that disables the possibility of a sensor going for ride-along and just makes users disable arming checks.

> Also, I remember that if airspeed is enable even not used by parameter it's used for wind estimation which helps a lot in maneuvering. Also it's used if just enabled in landing approach.

If that is true, I consider it a bug - if a sensor is set for not use, it shouldn't be used.

Reply:
You're right. I was confused by function name ahrs->airspeed_sensor_enabled() . Which I found that it checks for use() not just enabled().

Reply:
> Can you give an example of such sensor?

GPS2 is checked even if not used for example when GPS_AUTO_SWITCH = 0 also checked by GPS::all_consistent();
Non-primary compasses are checked by Compass::consistent() while only primary is actually used.

Reply:
> GPS2 is checked even if not used for example when GPS_AUTO_SWITCH = 0 also checked by GPS::all_consistent();

GPS is a special case in the sense that GPS_AUTO_SWITCH wasn't meant to be a way to select a primary.

> Non-primary compasses are checked by Compass::consistent() while only primary is actually used.

No, any compass that is set as to not use, won't be checked in the consistent method.